pandas/core/groupby/grouper.py

```diff
@@ -607,11 +607,9 @@ def is_in_axis(key) -> bool:
 
     # if the grouper is obj[name]
     def is_in_obj(gpr) -> bool:
-        if not hasattr(gpr, "name"):
-            return False
         try:
-            return gpr is obj[gpr.name]
-        except (KeyError, IndexError):
+            return id(gpr) == id(obj[gpr.name])
```

Comment:
how is this different?

Reply:
just reverting the change from the original PR as opening gambit. plan to backport so either the old code or minimal changes works for me.